src/dc_receive_imf.rs

```diff
@@ -1004,17 +1004,16 @@ fn create_or_lookup_group(
             }
         }
     }
-    if mime_parser.group_avatar != AvatarAction::None {
+    if let Some(avatar_action) = &mime_parser.group_avatar {
         info!(context, "group-avatar change for {}", chat_id);
         if let Ok(mut chat) = Chat::load_from_db(context, chat_id) {
-            match &mime_parser.group_avatar {
+            match avatar_action {
                 AvatarAction::Change(profile_image) => {
                     chat.param.set(Param::ProfileImage, profile_image);
                 }
                 AvatarAction::Delete => {
                     chat.param.remove(Param::ProfileImage);
                 }
-                AvatarAction::None => {}
```

Comment:
And here we don't have to consider `None` case, which actually should have been marked as `unreachable!`.